docs/reference/enable-gpu.rst

```diff
@@ -1,46 +1,153 @@
+created:
+
+
+.. code-block:: bash
+
+  while ! sudo microk8s.kubectl get ds -n gpu-operator-resources nvidia-operator-validator
+  do
+    sleep 5
+  done
+
+.. note::
+   It will take some seconds for the DaemonSet to get created. The above command will
+   return a message like the following at the beginning:
+   ``Error from server (NotFound): daemonsets.apps "nvidia-operator-validator" not found``
+
+The expected output should be:
+
+.. code-block:: text
+
+   Error from server (NotFound): daemonsets.apps "nvidia-operator-validator" not found
+   ...
+   Error from server (NotFound): daemonsets.apps "nvidia-operator-validator" not found
+   NAME                        DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR                                   AGE
+   nvidia-operator-validator   1         1         0       1            0           nvidia.com/gpu.deploy.operator-validator=true   17s
+
+Ensure the Validator succeeded
+""""""""""""""""""""""""""""""
+
+Next we'll need to wait for the Validator Pod to succeed:
+
+.. code-block:: bash
+
+  echo "Waiting for the NVIDIA Operator validations to complete..."
+  while ! sudo microk8s.kubectl logs -n gpu-operator-resources -l app=nvidia-operator-validator -c nvidia-operator-validator | grep "all validations are successful"
+  do
+    sleep 5
+  done
+
+.. note::
+   It will take some seconds for the Pod to get initialized . The above command will
+   return a message like the following at the beginning:
+   ``Error from server (BadRequest): container "nvidia-operator-validator" in pod "nvidia-operator-validator-4rq5n" is waiting to start: PodInitializing``
+
+The expected output should be:
+
+.. code-block:: text
+
+   Error from server (BadRequest): container "nvidia-operator-validator" in pod "nvidia-operator-validator-4rq5n" is waiting to start: PodInitializing
+   ...
+   Error from server (BadRequest): container "nvidia-operator-validator" in pod "nvidia-operator-validator-4rq5n" is waiting to start: PodInitializing
+   all validations are successful
+
+Verify the DSS detects the GPU
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: bash
+
+  dss status
+
+Expected output:
+
+.. code-block:: bash
+
+  [INFO] MLflow deployment: Ready
+  [INFO] MLflow URL: http://10.152.183.74:5000
+  [INFO] GPU acceleration: Enabled (NVIDIA-GeForce-RTX-3070-Ti)
 
-* NVIDIA drivers installed on your machine.
-* The MicroK8s runtime configured to be able to use the Nvidia drivers.
+.. note::
 
-The `MicroK8s GPU add-on`_ is installing the NVIDIA Operator. The NVIDIA operator can either install or detect the NVIDIA drivers if they are already present on the host machine.
+  The GPU model `NVIDIA-GeForce-RTX-3070-Ti` will be different depending on your device.
 
-Before installing DSS:
+Launch GPU-enabled Notebook
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-1. Follow the MicroK8s documentation to `Install MicroK8s`_.
+At this point the DSS is fully configured to utilise the host's GPU. The next step will
```

Comment:
It's important to understand that a how-to guide is not step by step guide. Each section are independent, so sentences like "at this point", "next" should be generally avoided.

Reply:
@afgambin I would disagree here. The step by step should matter, otherwise I would first tell users to create a Notebook with a GPU and then to install the drivers.

Maybe in what you describe would make sense in a case where I'd have one how-to for multiple orthogonal items (like starting, stoping, deleting notebooks). But in this case, to end up spinning up a Notebook and use a GPU the order matters

Reply:
hmm.. I see your point, but the how-to guides should generally be structured as orthogonal and independent items as much as possible, e.g., the user already installed the NVIDIA Operator, and now he wants to launch a GPU-enabled notebook -> so he goes "at work" and looks for that section (Launch a GPU-enable notebook).

And this is important to understand, when he lands at the section he is looking for, he already knows that to launch the notebook, the NVIDIA Operator must be already installed. Otherwise, we are writing a tutorial.

We can iterate on this, but the current text feels yet like writing a tutorial to me. It might be this is what we actually need - I remember you mention having a second tutorial involving GPUs.

Reply:
Ah, I see what you mean. In this case if we'd want to keep a how-to then does it mean we need to:
1. Have a `Enable NVIDIA GPUs in DSS` how-to, that handles installing and verifying the NVIDIA Operator
2. Have a `Launch a GPU-enabled notebook` how-to, that are prerequisite has the previous guide, and shows how to run the notebook

??

Don't have a hard preference on having it as a tutorial or how to, more on what's the most time effective approach that we can justify with the Diataxis framework

Reply:
Exactly. And they can still be in the same how-to page (because they are related). The key part is that they are content independent sections as much as possible and we use proper linking and text. For instance, in this case, what I would do is the following (for the `Launch a GPU-enabled notebook` section):

1. Modify the text to comply with how-to style guide -> "You can deploy a notebook containing CUDA runtimes and ML frameworks." -> straight to the action. No prior context.
2. Add a note below that intro sentence saying that: "To launch a GPU-enabled notebook, you must first install and verify the NVIDIA Operator", adding links to these sections in "install" and "verify" words.

In this way, there is no longer sequential and contextual text, therefore it is not a tutorial. I think with that I would be happy with the current state.

Reply:
ACK, thanks for the explanation! Pushed a commit for this 083f79a